pkg/apis/argoproj/v1alpha1/argocd_types.go

```diff
@@ -316,9 +316,21 @@ type ArgoCDServerServiceSpec struct {
 	Type corev1.ServiceType `json:"type"`
 }
 
+// ArgoCDScope defines the scope of resources that ArgoCD would be managing
+type ArgoCDScope struct {
+	// Cluster when enabled lets ArgoCD manage cluster resources.
+	Cluster bool `json:"cluster,omitempty"`
```

Comment:
How about rename to ClusterConfigEnabled since it is a bool?

Reply:
Made the required change.